### PR TITLE
redo(ticdc): changefeed global resolved timestamp should never fallback

### LINKED ISSUE #6456: redo log failed: [ERROR] [mysql.go:798] ["execute DMLs failed"] [err="[CDC:ErrMySQLTxnError]MySQL txn error: Error 1049: Unknown database 'workload'"]
### What did you do?

- v6.2.0 redo log failed, and can be reproduced.

- Failed case: 

  [redo_apply_cdc_allnode_restart_sync]
  [redo_enable_cdc_allnode_restart_sync]

### What did you expect to see?

_No response_

### What did you see instead?
ERROR log: 
[2022/07/23 19:21:54.430 +00:00] [ERROR] [mysql.go:798] ["execute DMLs failed"] [err="[CDC:ErrMySQLTxnError]MySQL txn error: Error 1049: Unknown database 'workload'"]
[2022/07/23 06:21:08.001 +00:00] [ERROR] [mysql.go:254] ["mysqlSink encountered error"] [error="[CDC:ErrMySQLTxnError]MySQL txn error: Error 1146: Table 'workload.sbtest5' doesn't exist"]

Execute plan id: 992369, 1021601 (this has extra failed case: [redo_enable_disable_basic_real_time_sync], [redo_disable_enable_basic_real_time_sync])

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
(paste TiCDC version here)
 [release-version=v6.2.0] [git-hash=934824b8c065e1a14c840e5927fac48b83289576]
```

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What problem does this PR solve?

Issue Number: close #6456 

### What is changed and how it works?

* Ensures changefeed global resolved ts never fallback
* After a table is removed, `RedoManager.GetMinResolvedTs` should be changed if necessary

#### Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by nongfushanquan — COMMENTED** [review]
please add some comments to  describe the root cause of the bug

File: cdc/owner/changefeed.go [hunk 1]
```diff
@@ -341,6 +345,13 @@ func (c *changefeed) tick(ctx cdcContext.Context, state *orchestrator.Changefeed
 				newResolvedTs = prevResolvedTs
 			}
 		}
+		// checkpointTs and resolvedTs should never regress.
+		if newResolvedTs < prevResolvedTs {
```
Comment (nongfushanquan) [inline cdc/owner/changefeed.go:349]: when will this case happen?

File: cdc/redo/manager.go [hunk 2]
```diff
@@ -388,6 +394,16 @@ func (m *ManagerImpl) RemoveTable(tableID model.TableID) {
 	if _, ok := m.rtsMap.LoadAndDelete(tableID); !ok {
 		log.Warn("remove a table not maintained in redo log manager", zap.Int64("tableID", tableID))
 	}
+	newMin := uint64(math.MaxInt64)
```
Comment (nongfushanquan) [inline cdc/redo/manager.go:397]: avoid the case which the table removed is the smallest resolvents?

File: cdc/owner/changefeed.go [hunk 3]
```diff
@@ -627,7 +648,7 @@ func (c *changefeed) handleBarrier(ctx cdcContext.Context) (uint64, error) {
 	barrierTp, barrierTs := c.barriers.Min()
 	phyBarrierTs := oracle.ExtractPhysical(barrierTs)
 	c.metricsChangefeedBarrierTsGauge.Set(float64(phyBarrierTs))
-	blocked := (barrierTs == c.state.Status.CheckpointTs) && (barrierTs == c.state.Status.ResolvedTs)
+	blocked := barrierTs == c.state.Status.CheckpointTs
```
Comment (nongfushanquan) [inline cdc/owner/changefeed.go:651]: Should syncpoint barrier (Line 656) also be changed ?

**Review by nongfushanquan — APPROVED** [review]

File: cdc/processor/pipeline/sink.go [hunk 4]
```diff
@@ -125,32 +125,27 @@ func (n *sinkNode) flushSink(ctx context.Context, resolved model.ResolvedTs) (er
...
+	// Flush sink with barrierTs, which is broadcasted by owner.
+	currentBarrierTs := atomic.LoadUint64(&n.barrierTs)
+	if resolved.Ts > currentBarrierTs {
+		resolved = model.NewResolvedTs(currentBarrierTs)
+	}
+	if n.redoManager != nil && n.redoManager.Enabled() {
 		redoTs := n.redoManager.GetMinResolvedTs()
-		if redoTs < currentBarrierTs {
+		if currentBarrierTs > redoTs {
+			// FIXME: How can barrierTs be greater than todoTs?
```
Comment (asddongmen) [inline cdc/processor/pipeline/sink.go:143]: typo

File: cdc/processor/pipeline/sink.go [hunk 5]
```diff
@@ -125,32 +125,27 @@ func (n *sinkNode) flushSink(ctx context.Context, resolved model.ResolvedTs) (er
...
 			log.Warn("redoTs should not less than current barrierTs",
 				zap.Int64("tableID", n.tableID),
 				zap.Uint64("redoTs", redoTs),
 				zap.Uint64("barrierTs", currentBarrierTs))
 		}
-
-		// TODO: remove this check after SchedulerV3 become the first choice.
-		if resolved.Ts > redoTs {
-			resolved = model.NewResolvedTs(redoTs)
-		}
```
Comment (3AceShowHand) [inline cdc/processor/pipeline/sink.go:149]: how about do not remove this, but use config to control it ? like https://github.com/pingcap/tiflow/blob/29d1882b691a5497a75041366ebc22bbefaee5d6/cdc/processor/processor.go#L270-L273

**Review by 3AceShowHand — APPROVED** [review]

File: cdc/owner/changefeed.go (hunk 1 above)
Comment (hicqu) [reply cdc/owner/changefeed.go:354]: Whe the owner is just initialized. It loads DDL starts at `checkpointTs-1`, which means resolvedTs can regress.

**Pull request merged**